### PR TITLE
Usage and some other Fixes & Tweaks

### PR DESCRIPTION
**Pull request by binaryoverload**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] All changes have been tested

## Added
 - Dynamic Usage System
 - An "extra info" for commands for display under usage (only used by welcome as of now)

## Changed
 - Purge command no longer has the "all" argument and instead takes a number as the first arg
 - Refactor *all* usages to work with the new system

## Deprecated
N/A

## Removed
N/A

## Fixed
 - Ryan's Code in permissions being repetitive and unreliable
 - Usage now uses the embed description to stop char overflows

## Security
N/A

### CODE REVIEW

**Comment by binaryoverload** [comment]
@weeryan17 I would appreciate a review from you as well :)

File: src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java [hunk 1]
```diff
@@ -17,7 +15,7 @@
 import stream.flarebot.flarebot.util.GeneralUtils;
 import stream.flarebot.flarebot.util.MessageUtils;
 
-import java.awt.Color;
+import java.awt.*;
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: I see our good old friend `java.awt.*` came back

File: src/main/java/stream/flarebot/flarebot/commands/music/RandomCommand.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public String getDescription() {
 
     @Override
     public String getUsage() {
-        return "{%}random [amount]";
+        return "`{%}random [amount]`";
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/commands/music/RandomCommand.java]: I know it's disabled but we should still provide more usage info here

File: src/main/java/stream/flarebot/flarebot/commands/music/SkipCommand.java [hunk 3]
```diff
@@ -163,8 +163,9 @@ public String getDescription() {
 
     @Override
     public String getUsage() {
-        return "{%}skip [yes/no]\n" +
-                "{%}skip force";
+        return "`{%}skip` - Starts a vote to skip the song\n" +
+                "`{%}skip yes|no` - Vote yes or no to skip the current song\n" +
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/commands/music/SkipCommand.java]: We should tell them that this is only used when a vote is running and to start a vote run `_skip`

File: src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java (hunk 1 above)
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/PermissionsCommand.java]: Fuck

File: src/main/java/stream/flarebot/flarebot/commands/music/RandomCommand.java (hunk 2 above)
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/commands/music/RandomCommand.java]: Sure

File: src/main/java/stream/flarebot/flarebot/commands/music/SkipCommand.java (hunk 3 above)
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/commands/music/SkipCommand.java]: Read the first line, it tells them how to start a vote. The error messages in the commands explain the rest

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/commands/music/SkipCommand.java]: Fair enough

File: src/main/java/stream/flarebot/flarebot/util/UsageParser.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                    for (String cmd : entry.getValue().getValue().split("\\|")) {
+                        if (args[entry.getKey()].equalsIgnoreCase(cmd)) {
+                            applicable = true;
+                            valid = true;
+                            break;
+                        }
+                    }
+                    if (!valid) {
+                        applicable = false;
+                        break; // CHeck nothing else if this fails
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/util/UsageParser.java]: *Check

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -24,3 +24,13 @@ A lot of the changes requested in PRs is related to memory usage and efficiency.
...
+### Usage Formatting
+###### If you don't follow these instructions when creating a new command it will automatically be denied
+The command usage now has the following format:
+ - The command (not the description) needs to be enclosed in graves \`.
+ - `[optional_argument]` for arguments that can be included but are not required.
+ - `<required_argument>` for arguments that are required.
+ - `subcommand` sub commands **must not** be enclosed in anything.
+ - `command|command1|command2` multiple sub commands not enclosed by anything and separated by a pipe.
+ - Spaces are only permitted between arguments.
+ - `\`{%}command\`` is the format for commands
```
Comment (WalshyDev) [inline CONTRIBUTING.md]: This part is broken, make sure to check the preview when editing Markdown
https://bots.are-pretty.sexy/3c5819.png

File: src/main/java/stream/flarebot/flarebot/commands/automod/AutoModCommand.java [hunk 6]
```diff
@@ -137,11 +137,11 @@ public String getDescription() {
...
-        return "{%}automod enable/disable - Enable or disable automod for your server!\n"
-                + "{%}automod config - View config/punishments for a server\n"
-                + "{%}automod punishments [reset] - View punishments. Reset punishments with the optional argument\n"
-                + "{%}automod punishments set <punishment> <points> [duration] - Description\n"
-                + "{%}automod whitelist list/add/remove [whitelist item] - Description\n"
+        return "{%}automod enable|disable - Enable or disable automod for your server!\n"
+                + "{%}automod config - View config/punishments for a server.\n"
+                + "{%}automod punishments [reset] - View punishments. Reset punishments with the optional argument.\n"
+                + "{%}automod punishments set <punishment> <points> [duration] - Description.\n"
+                + "{%}automod whitelist list|add|remove [whitelist_item] - Description.\n"
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/commands/automod/AutoModCommand.java]: ummmm lets see if you can spot the problem hint: `

File: src/main/java/stream/flarebot/flarebot/util/UsageParser.java (hunk 4 above)
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/util/UsageParser.java]: I swear

**Pull request merged**